### PR TITLE
Fix Snap Server Account Range tests

### LINKED ISSUE #7373: Snapsync server AccountRange hive tests
Besu snapsync server AccountRange tests are failing in hive.

### PR DESCRIPTION
**Pull request by Gabriel-Trintinalia**
## PR description
created on top of https://github.com/hyperledger/besu/pull/7535. 

## Fixed Issue(s)
Fixes #7373 and #6995

### CODE REVIEW

File: ethereum/core/src/main/java/org/hyperledger/besu/ethereum/worldstate/WorldStateArchive.java [hunk 1]
```diff
@@ -64,4 +64,6 @@ <U> Optional<U> getAccountProof(
       final Address accountAddress,
       final List<UInt256> accountStorageKeys,
       final Function<Optional<WorldStateProof>, ? extends Optional<U>> mapper);
+
+  WorldStateKeyValueStorage getWorldStateStorage();
```
[the remark this answers is not in the document]
Comment (Gabriel-Trintinalia) [reply ethereum/core/src/main/java/org/hyperledger/besu/ethereum/worldstate/WorldStateArchive.java]: done

File: ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/manager/snap/SnapServer.java [hunk 2]
```diff
@@ -240,12 +241,9 @@ MessageData constructGetAccountRangeResponse(final MessageData message) {
...
                         (pair) -> {
-                          var rlpOutput = new BytesValueRLPOutput();
-                          rlpOutput.startList();
-                          rlpOutput.writeBytes(pair.getFirst());
-                          rlpOutput.writeRLPBytes(pair.getSecond());
-                          rlpOutput.endList();
-                          return rlpOutput.encodedSize();
+                          Bytes bytes =
+                              AccountRangeMessage.toSlimAccount(RLP.input(pair.getSecond()));
+                          return Hash.SIZE + bytes.size();
```
[the remark this answers is not in the document]
Comment (Gabriel-Trintinalia) [reply ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/manager/snap/SnapServer.java:246]: Which other calculations?

File: ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/manager/snap/SnapServer.java [hunk 3]
```diff
@@ -257,11 +255,15 @@ MessageData constructGetAccountRangeResponse(final MessageData message) {
                     storage.streamFlatAccounts(range.startKeyHash(), shouldContinuePredicate);
 
                 if (accounts.isEmpty() && shouldContinuePredicate.shouldContinue.get()) {
+                  var fromNextHash =
+                      range.endKeyHash().compareTo(range.startKeyHash()) >= 0
```
[the remark this answers is not in the document]
Comment (Gabriel-Trintinalia) [reply ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/manager/snap/SnapServer.java:259]: test14_RequestStartingHashAfterLimitHash

File: ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/messages/snap/GetAccountRangeMessage.java [hunk 4]
```diff
@@ -48,12 +48,20 @@ public static GetAccountRangeMessage readFrom(final MessageData message) {
 
   public static GetAccountRangeMessage create(
       final Hash worldStateRootHash, final Bytes32 startKeyHash, final Bytes32 endKeyHash) {
+    return create(worldStateRootHash, startKeyHash, endKeyHash, SIZE_REQUEST);
+  }
+
+  public static GetAccountRangeMessage create(
```
[the remark this answers is not in the document]
Comment (Gabriel-Trintinalia) [reply ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/messages/snap/GetAccountRangeMessage.java:56]: done

File: ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/messages/snap/AccountRangeMessage.java [hunk 5]
```diff
@@ -131,6 +132,26 @@ private Bytes toFullAccount(final RLPInput rlpInput) {
     return rlpOutput.encoded();
   }
 
+  public static Bytes toSlimAccount(final RLPInput rlpInput) {
```
[the remark this answers is not in the document]
Comment (Gabriel-Trintinalia) [reply ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/messages/snap/AccountRangeMessage.java:135]: sure, will do

File: ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/messages/snap/AccountRangeMessage.java [hunk 6]
```diff
@@ -131,6 +132,26 @@ private Bytes toFullAccount(final RLPInput rlpInput) {
...
+    StateTrieAccountValue accountValue = StateTrieAccountValue.readFrom(rlpInput);
+    var rlpOutput = new BytesValueRLPOutput();
+    rlpOutput.startList();
+    rlpOutput.writeLongScalar(accountValue.getNonce());
+    rlpOutput.writeUInt256Scalar(accountValue.getBalance());
+    if (accountValue.getStorageRoot().equals(Hash.EMPTY_TRIE_HASH)) {
+      rlpOutput.writeNull();
```
[the remark this answers is not in the document]
Comment (Gabriel-Trintinalia) [reply ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/messages/snap/AccountRangeMessage.java:142]: you are right

File: ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/messages/snap/AccountRangeMessage.java (hunk 5 above)
[the remark this answers is not in the document]
Comment (Gabriel-Trintinalia) [reply ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/messages/snap/AccountRangeMessage.java:137]: done

File: ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/messages/snap/AccountRangeMessage.java (hunk 6 above)
[the remark this answers is not in the document]
Comment (Gabriel-Trintinalia) [reply ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/messages/snap/AccountRangeMessage.java:144]: The spec says `empty list` but the client implementations are Empty Byte Array

**Review by jframe — APPROVED** [review]

**Pull request merged**